File: lab3/lineService.cpp
//
// Created by kachm on 24.02.2022.
//
#include <iostream>
#include <vector>
#include "Line.h"
#include "linesService.h"

Line *createLines(int linesCount) {
    Line *lines = new Line[linesCount];

    for (int i = 0; i < linesCount; ++i) {
        int a, b, c;
        cout << "Input parameter [a]:";cin >> a;
        cout << "Input parameter [b]:";cin >> b;
        cout << "Input parameter [c]:";cin >> c;
        lines[i] = Line(a, b, c);
        cout << endl;
    }

    return lines;
}

Line *createLinesRandom(int linesCount) {
    Line *lines = new Line[linesCount];
    for (int i = 0; i < linesCount; ++i) {
        lines[i] = Line(0+(rand()%100),0+(rand()%100),0+(rand()%100));
    }
    return lines;
}

void getLinesIndexes(Line *lines, int linesCount, int xPos, int yPos) {
    for (int i = 0; i < linesCount; ++i) {
        if (lines[i].isLineContainsPoint(xPos, yPos)) {
            cout << "Line with index=" << i << " contains Point(" << xPos << "," << yPos << ")" << endl;
            lines[i].printLine();
        }
    }
}

File: python/lab3/LineService.py
from random import randrange
from Line import Line

def generateLinesManual():  # Генерування масиву прямих
    lines = []  # Створення масиву ліній
    for i in range(0, int(input('Input number of lines:'))):
        lines.append(Line(int(input('Input parameter [a]:')),  # Створення та ініціалізація обєкту
                          int(input('Input parameter [b]:')),
                          int(input('Input parameter [c]:'))))
    return lines

def generateLinesRandom():  # Генерування масиву прямих
    lines = []  # Створення масиву ліній
    for i in range(0, int(input('Input number of lines:'))):  # Цикл по кількості ліній
        lines.append(Line(randrange(50), randrange(200), randrange(100)))  # Створення та ініціалізація обєкту
    return lines
